src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -477,10 +533,25 @@ case class IndexLogEntry(
     sourcePlanSignatures.head
   }
 
-  def hasLineageColumn(spark: SparkSession): Boolean = {
-    ResolverUtils
-      .resolve(spark, IndexConstants.DATA_FILE_NAME_COLUMN, schema.fieldNames)
-      .isDefined
+  def hasLineageColumn: Boolean = schema.fieldNames.contains(IndexConstants.DATA_FILE_NAME_ID)
+
+  @JsonIgnore
+  lazy val fileNameToIdMap: Map[String, Long] = {
+    val fileNameToIdMap = mutable.Map[String, Long]()
+    sourceFileInfoSet.foreach(f => fileNameToIdMap.put(f.name, f.id))
```

Comment:
I think we should not add deleted/appended files here:
1. A file gets an id when it is indexed. So appendedFiles do not have an id yet. If we add them to the map, once we index them during refresh, since they are already in the map, their id remains -1L.
2. A deletedFile should already be in the current Content as well.  So, it should be already listed in the Map.

@sezruby  Do you agree?

Reply:
1 => how about assigning a new id to appended files? as I commented around `currentFiles`? 
- pros: all files have the id, appended files in a persisted log file also have file ids. A user might have some doubts for -1 id fields.
- cons: multiple id assignment points

2 => I agree, thanks :)

Reply:
As you mentioned in the cons, right now the logic of generating/assigning ids is in one place which is `Directory.fromLeafFiles` and that is clean and straightforward. I am not sure what would be the use case of assigning id to a file which is not indexed yet. Now that the changes are made to assign ids when creating FileInfo for source content, I prefer to keep the id assignment logic in one place for simplicity. 
I dont think we should be worry about user confusion, as Index Metadata is really for Hyperspace use only and users' application should not rely on that. Moreover, there is a good reason about why the file has an id of -1L. It is because it is not indexed yet.

Reply:
That makes sense. Let's keep that approach. Thanks :) 

Could you also check https://github.com/microsoft/hyperspace/pull/234#discussion_r517004751?